src/app/(proper_react)/layout.tsx

```diff
@@ -24,6 +25,24 @@ export default async function Layout({ children }: { children: ReactNode }) {
     email: session?.user.email ?? "",
   });
 
+  const cookieStore = cookies();
+  const gaCookie = cookieStore.get("_ga");
+
+  if (gaCookie && gaCookie.value) {
+    const [gaCookieVersion, gaCookiePath, gaCookieClientId, gaCookieTimestamp] =
+      gaCookie.value.split(".");
+
+    if (session?.user.subscriber?.id) {
+      await addClientIdForSubscriber(
+        session?.user.subscriber?.id,
+        gaCookieVersion,
+        parseInt(gaCookiePath),
+        gaCookieClientId,
+        parseInt(gaCookieTimestamp),
+      );
+    }
+  }
+
```

Comment:
@Vinnl do you think this belongs in a server-only component like `PageLoadEvent` or is it better directly in here? Or is there some other approach I'm missing?